test/Inputs/clang-importer-sdk/usr/include/ObjCConcurrency.h

```diff
@@ -16,6 +16,8 @@
 
 -(void)doSomethingConflicted:(NSString *)operation completionHandler:(void (^)(NSInteger))handler;
 -(NSInteger)doSomethingConflicted:(NSString *)operation;
+-(void)server:(NSString *)name restartWithCompletionHandler:(void (^)(void))block;
+-(void)server:(NSString *)name atPriority:(double)priority restartWithCompletionHandler:(void (^)(void))block;
```

Comment:
Suggestion: I'm not sure what real-life APIs you're looking at, but at least rationalizing based on this example, it may make the most sense to translate these APIs by prepending the information to the base name at all times: `await` is really taking the place of the completion handler, and it lives on the opposite side of the expression as compared to a trailing closure.

```
   server:atPriority:restartWithCompletionHandler:
//                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ trailing closures are ergonomic

   await server.server("localhost", atPriorityRestart: 0.8)
// ^^^^^                                      ^^^^^^^ await ... wait for it ... restart
```

If the Objective-C importer were to translate these names by first logically shuffling this information to the front, then it seems to me like it'd be a consistently better result:

```
// Original
thing:frobnicateWithCompletionHandler:

// Rearrange to pull name information next to `await` use site
frobnicateWithCompletionHandler:thing:

// Eliminate "WithCompletionHandler"
frobnicateThing:
```

When the design for async/await is eventually shared, I hope you'll devote some text to translation from legacy APIs for discussion and possible enhancement, just like the review process that the original Objective-C-to-Swift translation went through.